### ISSUE TITLE
RegExp constructor to support copying other regex objects

### ISSUE THREAD

**Issue by brettz9** [babel/babel#5978]
Choose one: is this a bug report or feature request? _Feature request_

### Input Code

```js
var regex = /something/;
var newRegex = new RegExp(regex, 'g');
```

### Expected Behavior

The regex should be copied without error as per ES6 behavior.

### Current Behavior

An error is thrown in older browsers.

### Possible Solution

Redefine `RegExp` to allow copying of regex objects.

### Context

User reported our post-Babel-transform IndexedDB was still erring due to this not being polyfilled.

### Your Environment

User reported "Safari 6 esqueue" browser.

**Comment by jridgewell** [comment]
This is a core-js issue, not Babel's. We don't polyfill behavoir, only static syntax. /cc @zloirock

**Comment by brettz9** [comment]
Ah, yes. And it looks like they already have [code to do this](https://github.com/zloirock/core-js/blob/master/modules/es6.regexp.constructor.js#L22-L28) but I don't see this `es6.regexp.constructor.js` file as part of https://github.com/babel/babel/tree/7.0/packages/babel-polyfill/src/core-js/modules (assuming I'm piecing things together correctly).

**Comment by jridgewell** [comment]
Then that's an error on our part. Want to add it to the list?

**Comment by brettz9** [comment]
Sorry not sure what you mean--do you mean submit a PR which includes the file?

**Comment by jridgewell** [comment]
Yup.

**Comment by brettz9** [comment]
In comparing https://github.com/zloirock/core-js/tree/master/modules and https://github.com/babel/babel/tree/7.0/packages/babel-polyfill/src/core-js/modules , there appear to be a good number of files missing, even when just comparing those beginning with "es6".

While I can provide a PR for just the missing RegExp constructor file (assuming there is a build step to auto-incorporate it), I am wondering whether you might be able to employ some other mechanism to ensure you remain in sync.

**Comment by existentialism** [comment]
The modules located in babel-polyfill/src/core-js are actually used by babel-preset-env and generated via script (ref: https://github.com/babel/babel/pull/5584).

babel-polyfill itself uses the entire [core-js shim](https://github.com/zloirock/core-js/blob/master/shim.js): https://github.com/babel/babel/blob/7.0/packages/babel-polyfill/src/index.js#L6

**Comment by StabbarN** [comment]
I've verified that https://github.com/zloirock/core-js/blob/master/modules/es6.regexp.constructor.js (`core-js/fn/regexp/constructor`) isn't added to Babel polyfill when using babel-present-env with: 
- `useBuiltIns: 'entry'`
- `import '@babel/polyfill'`
- browser list: `iOS >= 9`

by catching the  error `RegExp constructor to support copying other regex objects`

Adding `import 'core-js/fn/regexp/constructor'` right after `import '@babel/polyfill'` solves it.

I've made a relatively small repo ( https://github.com/StabbarN/indexeddbshim-regexp/ ) with indexeddbshim to verify it ( https://github.com/axemclion/IndexedDBShim/issues/311 ).

What's the process to add it to Babel polyfill with browsers earlier than ios 10? 

It's needed on ios9 according to https://kangax.github.io/compat-table/es6/#ios9 (see row "Symbol.match, RegExp constructor" in "well-known symbols")

**Comment by zloirock** [comment]
@StabbarN it should be fixed in #7646
